sdk/search/Azure.AI.Search/src/SearchClientOptions.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.AI.Search
+{
+    /// <summary>
+    /// Provides the client configuration options for connecting to Azure
+    /// Cognitive Search.
+    /// </summary>
+    public class SearchClientOptions : ClientOptions
+    {
+        /// <summary>
+        /// The versions of Azure Cognitive Search supported by this client
+        /// library.  For more, see
+        /// <see href="https://docs.microsoft.com/en-us/azure/search/search-api-versions" />.
+        /// </summary>
+        public enum ServiceVersion
+        {
+            #pragma warning disable CA1707 // Identifiers should not contain underscores
+            /// <summary>
+            /// The 2019-05-06 version of the service that is Generally Available.
+            /// </summary>
+            V2019_05_06 = 1
```

Comment:
Nit: terminating comma keeps the diff down later.

Reply:
Those always feel like when somebody starts talking and 

I'm personally not a fan and don't mind an extra line of red in the diff.  If I'm the only one on the team who feels that way, I'll change it (and suggest we add a style rule).